rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/flow/TbRuleChainInputNode.java

```diff
@@ -48,17 +56,81 @@ public class TbRuleChainInputNode implements TbNode {
 
     private TbRuleChainInputNodeConfiguration config;
     private RuleChainId ruleChainId;
+    private boolean forwardMsgToDefaultRuleChain;
 
     @Override
     public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
         this.config = TbNodeUtils.convert(configuration, TbRuleChainInputNodeConfiguration.class);
-        this.ruleChainId = new RuleChainId(UUID.fromString(config.getRuleChainId()));
+        this.forwardMsgToDefaultRuleChain = config.isForwardMsgToDefaultRuleChain();
+        if (forwardMsgToDefaultRuleChain) {
+            this.ruleChainId = getRootRuleChainId(ctx);
+            return;
+        }
+        UUID ruleChainUUID;
+        try {
+            ruleChainUUID = UUID.fromString(config.getRuleChainId());
+        } catch (Exception e) {
+            throw new TbNodeException("Failed to parse rule chain id: " + config.getRuleChainId(), true);
+        }
+        if (ruleChainUUID.equals(ctx.getSelf().getRuleChainId().getId())) {
+            throw new TbNodeException("The rule chain id and the current rule chain id cannot be equals!", true);
+        }
+        this.ruleChainId = new RuleChainId(ruleChainUUID);
         ctx.checkTenantEntity(ruleChainId);
     }
 
     @Override
     public void onMsg(TbContext ctx, TbMsg msg) {
-        ctx.input(msg, ruleChainId);
+        RuleChainId targetRuleChainId = getRuleChainId(ctx, msg);
+        if (targetRuleChainId.equals(ctx.getSelf().getRuleChainId())) {
```

Comment:
if forwardMsgToDefaultRuleChain is false, we don't need to check for `targetRuleChainId.equals(ctx.getSelf().getRuleChainId())` since we checked that in the init method.